include/rtweekend.hpp

```diff
@@ -91,6 +91,18 @@ class LocalPseudoRNG {
   xorshift<> generator;
 };
 
+template<typename T>
```

Comment:
i gess task_context because of the complexity of accessor types.

maybe we could use get_pointer() and store a sycl::global_ptr<uint8_t> instead. this would remove the need to template it.

Reply:
Done in commit af750e9